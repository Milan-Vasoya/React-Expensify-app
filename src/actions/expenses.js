
import database from '../firebase/firebase';

//ADD expense
export const addExpense = (expense = {}) => ({

    type: 'ADD_EXPENSE',
    expense
});

export const startAddExpense = (expenseData = {}) => {
    return (dispatch, getState) => {
        const uid=getState().auth.uid;
        const { description = '',
            note = '',
            amount = 0,
            createdAt = 0 } = expenseData;

        const expense = { description, note, amount, createdAt };
        return database.ref(`users/${uid}/expenses`).push(expense).then((ref) => {
            dispatch(addExpense({
                id: ref.key,
                ...expense
            }));
        }
        )
    }
}

//remove expense
export const removeExpense = (id) => ({
    type: 'REMOVE_EXPENSE',
    id
});

export const startRemoveExpense = (id) => {
    return (dispatch, getState) => {
        const uid=getState().auth.uid;
        return database.ref(`users/${uid}/expenses/${id}`).remove().then(
            dispatch(removeExpense(id))
        );
    }
}
//Edit Expense
export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
});

export const startEditExpense = (id, expense) => {
    return (dispatch, getState) => {
        const uid=getState().auth.uid;
        return database.ref(`users/${uid}/expenses/${id}`).set(expense).then(
            dispatch(editExpense(id, expense))
        );
    }
}


//Set expenses
export const setExpenses = (expenses) => ({
    type: 'SET_EXPENSES',
    expenses
});



export const startSetExpense = () => {
    return (dispatch, getState) => {
        const uid=getState().auth.uid;

        return database.ref(`users/${uid}/expenses`).once('value').then((refs) => {
            const expenses = [];
            refs.forEach((ref) => {
                expenses.push({
                    id: ref.key,
                    ...ref.val()
                });
            });

            dispatch(setExpenses(expenses));

        });
    }
}