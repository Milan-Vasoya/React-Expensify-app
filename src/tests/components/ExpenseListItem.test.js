import React from 'react';
import {shallow} from 'enzyme' ;
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses  from '../fixture/expenses';

test('Should run Expense list Item', ()=>{
    const wrapper = shallow(< ExpenseListItem {...expenses[1]} />);
    expect(wrapper).toMatchSnapshot();
});

