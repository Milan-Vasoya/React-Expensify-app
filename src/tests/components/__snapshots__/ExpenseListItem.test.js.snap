// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should run Expense list Item 1`] = `
<div>
  <Link
    to="/edit/2"
  >
    <h3>
      Rent
    </h3>
  </Link>
  <p>
    Amount:
    109500
     is created AT: 
    -345600000
  </p>
</div>
`;
